src/shared/open-file.h

```diff
@@ -0,0 +1,34 @@
+
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include "list.h"
+
+typedef enum OpenFileFlags {
+        OPENFILE_READ_ONLY = 1 << 0,
+        OPENFILE_APPEND    = 1 << 1,
+        OPENFILE_TRUNCATE  = 1 << 2,
+        OPENFILE_GRACEFUL  = 1 << 3,
+        _OPENFILE_MAX,
+        _OPENFILE_INVALID  = -EINVAL,
+        _OPENFILE_MASK_PUBLIC = OPENFILE_READ_ONLY | OPENFILE_APPEND | OPENFILE_TRUNCATE | OPENFILE_GRACEFUL,
+} OpenFileFlags;
+
+typedef struct OpenFile {
+        char *path;
+        char *fdname;
+        uint64_t flags;
```

Comment:
Please use `OpenFileFlag` instead of `uint64_t`.

Reply:
`flags` can contain a combination of flags (i.e. `OPENFILE_READ_ONLY` and `OPENFILE_GRACEFUL`) so I don't think we can use the `OpenFileFlag` type for it.

Reply:
I think it'll work, we tend to use the enum type for flags even if the actual value can contain multiple flags.